profiling/src/internal/profile.rs

```diff
@@ -847,32 +838,37 @@ mod api_tests {
         assert_eq!(profile.sample_types, prev.sample_types);
 
         // The string table should have at least the empty string.
-        assert!(!profile.strings.is_empty());
+        assert!(profile.strings.len() > 0);
     }
 
     #[test]
     fn reset_period() {
         /* The previous test (reset) checked quite a few properties already, so
          * this one will focus only on the period.
          */
-        let sample_types: &[api::ValueType] = &[api::ValueType::new("wall-time", "nanoseconds")];
+        let sample_types = [api::ValueType::new("wall-time", "nanoseconds")];
         let period = api::Period {
             r#type: sample_types[0],
             value: 10_000_000,
         };
-        let mut profile = Profile::new(SystemTime::now(), sample_types, Some(period));
+        let mut profile = Profile::new(SystemTime::now(), &sample_types, Some(period));
 
         let prev = profile
             .reset_and_return_previous(None)
             .expect("reset to succeed");
 
         // Resolve the string values to check that they match (their string
         // table offsets may not match).
+        let mut strings = Vec::with_capacity(profile.strings.len());
+        let mut strings_iter = profile.strings.into_lending_iter();
```

Comment:
Could this become a `.map().collect()`?

Reply:
The lending iterator doesn't have it implemented. I could do that if you want, but I'd prefer to do it in another PR. Let me know.

Reply:
Doing `map` is possible it has usability issues. `collect` seems even harder, because you need to have a `Map` or something which breaks the borrow of the item (e.g. `String::from`) because otherwise the borrow prevents the lending iterator from moving to the next item (because it's `mut`).